### PR TITLE
Allow admin user to configure site languages on the portal

### PR DESCRIPTION
**Pull request by markoparfeniuk**
### Summary of changes:

- add models, repositories, services, validators and controllers to allow admin to configure languages;
- covered with tests.

### [Requirements](https://github.com/dark-side/lanthanum/tree/master/sports_hub_portal/web_application_features/manage_articles/user_stories/delete_existing_article)

### CODE REVIEW

**Review by foxerest — APPROVED** [review]
nice job

File: src/SportsHub.Business/Repositories/ILanguageRepository.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+        Task EditLanguageAsync(Language language);
+
+        Task<Language> DeleteLanguageAsync(Guid id);
+
+        Task<bool> DoesLanguageAlreadyExistByNameAsync(string languageName);
+
+        Task<bool> DoesLanguageAlreadyExistByCodeAsync(string languageCode);
+
+        Task<bool> DoesLanguageAlreadyExistByIdAsync(Guid id);
```
Comment (NotJediYet) [inline src/SportsHub.Business/Repositories/ILanguageRepository.cs:25]: I can suggest you better naming for these methods, I suppose.
```suggestion
        Task<bool> ExistsByIdAsync(Guid id);
```

Comment (NotJediYet) [reply src/SportsHub.Business/Repositories/ILanguageRepository.cs:25]: But it's simply a suggestion, If you want to, you can apply this renaming in separate PR for all methods

File: src/SportsHub.Business/Services/Implementation/LanguageService.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            var language = await _languageRepository.GetLanguageByNameAsync(languageName);
+
+            if (language == null)
+            {
+                return Guid.Empty;
+            }
+            else
+            {
+                return language.Id;
+            }
```
Comment (NotJediYet) [inline src/SportsHub.Business/Services/Implementation/LanguageService.cs:71]:
```suggestion
            return language.Id ?? Guid.Empty : 
```

Comment (NotJediYet) [reply src/SportsHub.Business/Services/Implementation/LanguageService.cs:71]: But do you really need this method?

File: src/SportsHub.Web/Validators/EditLanguageModelValidator.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        private async Task<bool> DoesLanguageCodeIsUniqueAsync(string languageCode, Guid Id)
+        {
+            var result = await _languageService.GetLanguageIdByCodeAsync(languageCode);
+
+            if (result == Guid.Empty)
+            {
+                return true;
+            }
+            else if (result == Id) { return true; }
+            else return false;
```
Comment (NotJediYet) [inline src/SportsHub.Web/Validators/EditLanguageModelValidator.cs]: if/else should always have curly braces. Fix it, please
```suggestion
            if (result == Guid.Empty)
            {
                return true;
            }
            else if (result == Id)
            {
                return true;
            }
            else
            {
                return false;
            }
```

**Review by NotJediYet — COMMENTED** [review]
Looks good. Please, fix the comments and you're free to merge it

File: src/SportsHub.Business/Services/Implementation/LanguageService.cs (hunk 2 above)
Comment (markoparfeniuk) [reply src/SportsHub.Business/Services/Implementation/LanguageService.cs:71]: Yes, it is used in validators.

File: src/SportsHub.Business/Repositories/ILanguageRepository.cs (hunk 1 above)
Comment (markoparfeniuk) [reply src/SportsHub.Business/Repositories/ILanguageRepository.cs:25]: I would keep it as it is. The naming is tied to the entity with which it operates.

**Pull request merged**